libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs

```diff
@@ -220,13 +232,13 @@ private string GetOsPath(string ambigiousPath)
 
         /// <summary>
         /// Parse lg content to LGTemplate list.
-        /// All the imports of the content will also be tracked down to parse templates.
+        /// All the imports of the content will be also tracked down to parse templates.
         /// </summary>
         /// <param name="content">Text content contains lg templates.</param>
         /// <param name="name">Text name.</param>
         /// <param name="importResolver">resolver to resolve LG import id to template text.</param>
         /// <returns>LGTemplate list of parsed lg content.</returns>
-        private IList<LGTemplate> ParseContent(string content, string name, ImportResolverDelegate importResolver)
+        private List<LGTemplate> ParseContent(string content, string name, ImportResolverDelegate importResolver)
         {
             var sources = new Dictionary<string, LGResource>();
             LoopLGText(content, name, sources, importResolver);
```

Comment:
We should have a pure resource discovery function like
Dict<string, LGResource> =  DiscoverLGResoures(LGResource start, Resolver importResolver); // already de-dup
which make this more decoupled and simplify the whole design

Reply:
Changed to DiscoverLGResources(LGResource start, ImportResolverDelegate importResolver, HashSet<string> parsedResoureceIds). Use the hashset to track visited files